Objects/obmalloc.c

```diff
@@ -2663,9 +2683,71 @@ _PyDebugAllocatorStats(FILE *out,
     (void)printone(out, buf2, num_blocks * sizeof_block);
 }
 
+int _PyMem_init_obmalloc(PyInterpreterState *interp, _PyRuntimeState *runtime)
+{
+#ifdef WITH_PYMALLOC
+    /* Initialize obmalloc, but only for subinterpreters,
+       since the main interpreter is initialized statically. */
+    if (interp == &runtime->_main_interpreter
+            || (interp->feature_flags & Py_RTFLAGS_USE_MAIN_OBMALLOC)) {
+        interp->obmalloc = &obmalloc_state_main;
+        interp->obmalloc->heap_allocated = false;
+    } else {
+        interp->obmalloc = PyMem_RawCalloc(1, sizeof(struct _obmalloc_state));
+        if (interp->obmalloc == NULL) {
+            return 0;
+        }
+        interp->obmalloc->heap_allocated = true;
+    }
+    if (!interp->obmalloc->initialized) {
+        // initialize the obmalloc->pools structure.  This must be done
+        // before the obmalloc alloc/free functions can be called.
+        poolp temp[OBMALLOC_USED_POOLS_SIZE] =
+            _obmalloc_pools_INIT(interp->obmalloc->pools);
+        memcpy(&interp->obmalloc->pools.used, temp, sizeof(temp));
+        interp->obmalloc->initialized = true;
+    }
+#endif /* WITH_PYMALLOC */
+    return 1;
```

Comment:
This feels more like a place we'd return -1 for "error" rather than a boolean for success.

```suggestion
            return -1;
        }
        interp->obmalloc->heap_allocated = true;
    }
    if (!interp->obmalloc->initialized) {
        // initialize the obmalloc->pools structure.  This must be done
        // before the obmalloc alloc/free functions can be called.
        poolp temp[OBMALLOC_USED_POOLS_SIZE] =
            _obmalloc_pools_INIT(interp->obmalloc->pools);
        memcpy(&interp->obmalloc->pools.used, temp, sizeof(temp));
        interp->obmalloc->initialized = true;
    }
#endif /* WITH_PYMALLOC */
    return 0;
```